### PR TITLE
ENH: add initial attenuator calculator devices - separate from the motion components for now

### PR DESCRIPTION
**Pull request by klauer**
Pairs with https://github.com/pcdshub/hxr-attenuator caproto IOC work

<details>
<summary>attempt 1</summary>
image
</details>

image

### CODE REVIEW

File: pcdsdevices/attenuator.py [hunk 1]
```diff
@@ -441,3 +443,245 @@ class GasAttenuator(Device, BaseInterface):
...
+    )
+    is_stuck = Cpt(
+        EpicsSignal, 'IsStuck', kind='hinted',
+        doc='Is the filter stuck / unusable?',
+    )
+    closest_energy = Cpt(
+        EpicsSignalRO, 'ClosestEnergy_RBV', kind='config',
+        doc='Closest tabulated energy available to the requested one',
+    )
+    transmission = Cpt(EpicsSignal, 'Transmission_RBV', kind='normal',
```
Comment (ZLLentz) [inline pcdsdevices/attenuator.py:465]: it's kind of cool you're doing this per-filter, the previous calculation IOCs didn't have this

File: pcdsdevices/attenuator.py [hunk 2]
```diff
@@ -441,3 +443,245 @@ class GasAttenuator(Device, BaseInterface):
...
+        EpicsSignal, ':SYS:ApplyConfiguration', kind='config',
+        doc='Apply the best configuration (i.e., move the filters)',
+    )
+    set_metadata(apply_config, dict(variety='command-proc', value=1))
+
+    moving = Cpt(
+        EpicsSignal, ':SYS:Moving_RBV', kind='config',
+        doc='Are filters being moved in/out?',
+    )
+    set_metadata(moving, dict(variety='bitmask', bits=1))
```
Comment (ZLLentz) [inline pcdsdevices/attenuator.py:599]: Does this give us a little light up box? If so it's a good trick I'd like to bring to other classes

Comment (klauer) [reply pcdsdevices/attenuator.py:599]: Yes:

image

(Although in pasting this, I realized I noticed that this should be `EpicsSignalRO` so it only generates the readback widget)

File: pcdsdevices/attenuator.py [hunk 3]
```diff
@@ -441,3 +443,245 @@ class GasAttenuator(Device, BaseInterface):
...
+
+    def get_best_config(self, **kwargs):
+        """Get the calculated (best) filter configuration."""
+        return self._bitmask_to_list(self.best_config_bitmask.get(**kwargs))
+
+    def get_moving_status(self, **kwargs):
+        """Get the filter motion status."""
+        return self._bitmask_to_list(self.filters_moving_bitmask.get(**kwargs))
+
+    def calculate(self, transmission, *, energy=None, use_floor=True):
```
Comment (ZLLentz) [inline pcdsdevices/attenuator.py:625]: :+1: very useful helper method

File: pcdsdevices/attenuator.py (hunk 2 above)
Comment (ZLLentz) [reply pcdsdevices/attenuator.py:599]: Still beats `moving      1`

**Comment by klauer** [comment]
I'll make an issue for combining this with the motion device, but will go with this version for the time being.

No major objections with the naming (however verbose and ugly I made it here), I hope?

**Comment by ZLLentz** [comment]
RE: the naming, the only thing that bugs me about the `attenuator` module naming in general is that some things are not verbose enough... There are several different implementations of attenuator IOCs and you have to read the docstrings to figure out which one you actually want. +1 to verbosity.

**Comment by klauer** [comment]
👍 Sounds good. Here we go...

**Pull request merged**